packages/@lwc/ssr-compiler/src/compile-template/element.ts

```diff
@@ -55,7 +61,15 @@ function yieldAttrOrPropLiteralValue(
 ): EsStatement[] {
     const { value, type } = valueNode;
     if (typeof value === 'string') {
-        const yieldedValue = name === 'style' ? normalizeStyleAttribute(value) : value;
+        let yieldedValue: string;
+        if (name === 'style') {
+            yieldedValue = normalizeStyleAttribute(value);
+        } else if (name === 'class') {
+            // @ts-expect-error weird indirection results in wrong overload being picked up
+            yieldedValue = StringReplace.call(StringTrim.call(value), /\s+/g, ' ');
```

Comment:
I don't think we're super strict about using these in `@lwc/ssr-compiler`, but maybe we should.